### PR TITLE
Custom project request properties (axon server context)

### PR DESCRIPTION
**Pull request by stefanvozd**
Adds custom project requests that can, based on request generate additional resources
Adds possibility to include Axon Server SE/EE/SAAS

### CODE REVIEW

File: initializr-be/src/main/resources/configuration/axon-server/saas/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## Running in Axon SaaS
+
+* Get free access to developer context at https://cloud.axoniq.io/
```
Comment (smcvb) [inline initializr-be/src/main/resources/configuration/axon-server/saas/README.md:3]:
```suggestion
* Get free access to a developer context at https://cloud.axoniq.io/
```

File: initializr-be/src/main/resources/configuration/axon-server/saas/README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+* Create application binding token and paste it in `src/main/resources/application.properties`
```
Comment (smcvb) [inline initializr-be/src/main/resources/configuration/axon-server/saas/README.md:4]:
```suggestion
* Create an application binding and paste the token in `src/main/resources/application.properties`
```

File: initializr-be/src/main/resources/configuration/axon-server/se/docker/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## Running Axon Server SE in docker-compose
+
+* `docker/docker-compose.yml` you can start Axon Server with the "`docker-compose up`" command.
```
Comment (smcvb) [inline initializr-be/src/main/resources/configuration/axon-server/se/docker/README.md:3]:
```suggestion
* The `docker/docker-compose.yml` allows you to start Axon Server SE with the "`docker-compose up`" command
```

File: initializr-be/src/main/resources/configuration/axon-server/saas/application.properties.mustache [hunk 4]
```diff
@@ -0,0 +1,8 @@
+####################################################################
+# Get free access to developer context at https://cloud.axoniq.io/ #
+####################################################################
+axon.axonserver.servers=axonserver.cloud.axoniq.io:443
+axon.axonserver.keep-alive-time=0
+axon.axonserver.ssl-enabled=true
+axon.axonserver.context={{axonServerContext}}
+axon.axonserver.token= #** your application binding token here **#
```
Comment (smcvb) [inline initializr-be/src/main/resources/configuration/axon-server/saas/application.properties.mustache:8]:
```suggestion
axon.axonserver.token= #** your application binding's token here **#
```

**Review by smcvb — COMMENTED** [review]
Bunch of nits, nothing major. Saw it working, so trusting in the outcome.

File: initializr-be/src/main/java/io/axoniq/initializr/customcontroller/AxonProjectRequestToDescriptionConverter.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @author Stefan Dragisic
+ */
+public class AxonProjectRequestToDescriptionConverter
+        implements ProjectRequestToDescriptionConverter<AxonProjectRequest> {
+
+    @Override
+    public ProjectDescription convert(AxonProjectRequest request, InitializrMetadata metadata) {
+        AxonProjectDescription description = new AxonProjectDescription();
+        new DefaultProjectRequestToDescriptionConverter().convert(request, description, metadata);
```
[the remark this answers is not in the document]
Comment (stefanvozd) [reply initializr-be/src/main/java/io/axoniq/initializr/customcontroller/AxonProjectRequestToDescriptionConverter.java:21]: Yes, its filled with only standard initilizer project fields (like app name, packange, depndecies etc), but we need to manually set our own fields

File: initializr-be/src/main/resources/configuration/axon-server/ee/docker/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,84 @@
+version: '3.1'
+services:
+  axonserver-enterprise-1:
+    image: axoniq/axonserver-enterprise:4.5.9-dev
```
Comment (lfgcampos) [inline initializr-be/src/main/resources/configuration/axon-server/ee/docker/docker-compose.yml:4]: Do we have a `latest` image tag? if we do, I would use that

File: initializr-be/src/main/resources/configuration/axon-server/ee/docker/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,84 @@
+version: '3.1'
```
Comment (lfgcampos) [inline initializr-be/src/main/resources/configuration/axon-server/ee/docker/docker-compose.yml:1]: I believe latest is `3.8` already

File: initializr-be/src/main/resources/configuration/axon-server/saas/README.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+## Running in Axon SaaS
```
Comment (lfgcampos) [inline initializr-be/src/main/resources/configuration/axon-server/saas/README.md:1]:
```suggestion
## Running in AxonIQ Cloud
```

File: initializr-be/src/main/resources/configuration/axon-server/se/docker/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
+version: '3.3'
```
Comment (lfgcampos) [inline initializr-be/src/main/resources/configuration/axon-server/se/docker/docker-compose.yml:1]: Same remark about version here

File: initializr-be/src/main/java/io/axoniq/initializr/customcontroller/AxonProjectContributor.java [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+/**
+ * A {@link ProjectContributor} generates additional project resources based on user request.
+ *
+ * @author Stefan Dragisic
+ */
+public class AxonProjectContributor implements ProjectContributor {
+
+    private static final Log logger = LogFactory.getLog(AxonProjectContributor.class);
+
+    private final ProjectDescription description;
```
[the remark this answers is not in the document]
Comment (stefanvozd) [reply initializr-be/src/main/java/io/axoniq/initializr/customcontroller/AxonProjectContributor.java:30]: thanks! its not needed at all

**Pull request merged**